src/linker/Linker.Steps/ProcessLinkerXmlStepBase.cs

```diff
@@ -0,0 +1,439 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using System.Xml.XPath;
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+	public abstract class ProcessLinkerXmlStepBase : BaseStep
+	{
+		const string FullNameAttributeName = "fullname";
+		const string LinkerElementName = "linker";
+		const string TypeElementName = "type";
+		const string SignatureAttributeName = "signature";
+		const string NameAttributeName = "name";
+		const string FieldElementName = "field";
+		const string MethodElementName = "method";
+		const string EventElementName = "event";
+		const string PropertyElementName = "property";
+		const string AllAssembliesFullName = "*";
+		protected const string XmlNamespace = "";
+
+
+		protected readonly string _xmlDocumentLocation;
+		readonly XPathDocument _document;
+		readonly EmbeddedResource _resource;
+		readonly AssemblyDefinition _resourceAssembly;
+
+		protected ProcessLinkerXmlStepBase (XPathDocument document, string xmlDocumentLocation)
+		{
+			_document = document;
+			_xmlDocumentLocation = xmlDocumentLocation;
+		}
+
+		protected ProcessLinkerXmlStepBase (XPathDocument document, EmbeddedResource resource, AssemblyDefinition resourceAssembly, string xmlDocumentLocation)
+			: this (document, xmlDocumentLocation)
+		{
+			_resource = resource ?? throw new ArgumentNullException (nameof (resource));
+			_resourceAssembly = resourceAssembly ?? throw new ArgumentNullException (nameof (resourceAssembly));
+		}
+
+		protected virtual bool ShouldProcessElement (XPathNavigator nav) => FeatureSettings.ShouldProcessElement (nav, Context, _xmlDocumentLocation);
+
+		protected virtual void ProcessXml (bool stripResource, bool ignoreResource)
+		{
+			try {
+				XPathNavigator nav = _document.CreateNavigator ();
+
+				// Initial structure check - ignore XML document which don't look like linker XML format
+				if (!nav.MoveToChild (LinkerElementName, XmlNamespace))
+					return;
+
+				if (_resource != null) {
+					if (stripResource)
+						Context.Annotations.AddResourceToRemove (_resourceAssembly, _resource);
+					if (ignoreResource)
+						return;
+				}
+
+				if (!ShouldProcessElement (nav))
+					return;
+
+				ProcessAssemblies (nav.SelectChildren ("assembly", ""));
+
+			} catch (Exception ex) when (!(ex is LinkerFatalErrorException)) {
+				throw new LinkerFatalErrorException (MessageContainer.CreateErrorMessage ($"Error processing '{_xmlDocumentLocation}'", 1013), ex);
```

Comment:
This is my fault, but the description of the error still talks about an XML descriptor, maybe better to change it to XML only

Reply:
I fixed the md file to only talk about XML.